scripts/detectNotUsingGitPush1by1.fsx

```diff
@@ -842,3 +844,416 @@ if notUsedGitPush1by1 then
+        "languages_url": "https://api.github.com/repos/realmarv/conventions/languages",
+        "stargazers_url": "https://api.github.com/repos/realmarv/conventions/stargazers",
+        "contributors_url": "https://api.github.com/repos/realmarv/conventions/contributors",
+        "subscribers_url": "https://api.github.com/repos/realmarv/conventions/subscribers",
+        "subscription_url": "https://api.github.com/repos/realmarv/conventions/subscription",
+        "commits_url": "https://api.github.com/repos/realmarv/conventions/commits{/sha}",
+        "git_commits_url": "https://api.github.com/repos/realmarv/conventions/git/commits{/sha}",
+        "comments_url": "https://api.github.com/repos/realmarv/conventions/comments{/number}",
+        "issue_comment_url": "https://api.github.com/repos/realmarv/conventions/issues/comments{/number}",
+        "contents_url": "https://api.github.com/repos/realmarv/conventions/contents/{+path}",
+        "compare_url": "https://api.github.com/repos/realmarv/conventions/compare/{base}...{head}",
+        "merges_url": "https://api.github.com/repos/realmarv/conventions/merges",
+        "archive_url": "https://api.github.com/repos/realmarv/conventions/{archive_format}{/ref}",
+        "downloads_url": "https://api.github.com/repos/realmarv/conventions/downloads",
+        "issues_url": "https://api.github.com/repos/realmarv/conventions/issues{/number}",
+        "pulls_url": "https://api.github.com/repos/realmarv/conventions/pulls{/number}",
+        "milestones_url": "https://api.github.com/repos/realmarv/conventions/milestones{/number}",
+        "notifications_url": "https://api.github.com/repos/realmarv/conventions/notifications{?since,all,participating}",
+        "labels_url": "https://api.github.com/repos/realmarv/conventions/labels{/name}",
+        "releases_url": "https://api.github.com/repos/realmarv/conventions/releases{/id}",
+        "deployments_url": "https://api.github.com/repos/realmarv/conventions/deployments"
+      }
+    }
+  ]
+}    
+""">
+
+prCommits
+|> Seq.iter(fun commit ->
+
+    let url =
+        sprintf
+            "https://api.github.com/repos/%s/commits/%s/check-suites"
+            gitRepo
+            commit
+
+    let checkSuitesJsonString = GitHubApiCall url
+
+    let checkSuitesParsedJson = CheckSuitesType.Parse checkSuitesJsonString
+
+    let commitMessageTitle =
+        (checkSuitesParsedJson.CheckSuites.[0]
+            .HeadCommit
+            .Message)
+            .Split(
+            "\n"
+        ).[0]
+
+    let status = checkSuitesParsedJson.CheckSuites.[0].Status
+    let conclusion = checkSuitesParsedJson.CheckSuites.[0].Conclusion
+
+    if
+        status = "completed" && conclusion = "failure"
+        && not(commitMessageTitle.Contains "failing test")
+    then
+        Console.Error.WriteLine
+            $"The CI status of a commit that doesn't add a failing test must be `success`."
```

Comment:
@realmarv nit: if you're not interpolating anything, there's no need for $ char. Anyway let's improve this message, I think it's better written like this: "Thanks for pushing commits 1 by 1, however we have detected some of them to not be successful (or not be red when they add a failing test)", which makes me think that this PR should also detect if a commit msg contains "failing test" and CI status is green, it should fail as well

Reply:
Done